sist02.csl

```diff
@@ -8,199 +8,133 @@
-            <date-part name="month" form="numeric-leading-zeros" range-delimiter="-" strip-periods="true" suffix="-"/>
-            <date-part name="day" form="numeric-leading-zeros"/>
-          </date>
-        </group>
-      </if>
-    </choose>
+    <group delimiter=", ">
+      <text variable="URL" prefix=" "/>
+      <group>
+        <text term="accessed" text-case="lowercase" prefix="(" suffix=" "/>
+        <date variable="accessed" suffix=")">
+          <date-part name="year" range-delimiter="-" suffix="-"/>
+          <date-part name="month" form="numeric-leading-zeros" range-delimiter="-" strip-periods="true" suffix="-"/>
+          <date-part name="day" form="numeric-leading-zeros"/>
+        </date>
+      </group>
+    </group>
   </macro>
   <macro name="page">
     <choose>
-      <if type="article-journal" match="any">
-        <label plural="never" suffix=" " variable="page" form="short"/>
-        <text variable="page" suffix="."/>
-      </if>
-      <else-if type="book" match="any">
+      <if type="article-journal chapter" match="any">
+        <label plural="never" suffix=". " variable="page" form="short"/>
         <text variable="page"/>
-        <choose>
-          <if match="any" variable="collection-title">
-            <label plural="never" suffix=", " variable="page" form="short"/>
-          </if>
-          <else>
-            <label plural="never" suffix="." variable="page" form="short"/>
-          </else>
-        </choose>
+      </if>
+      <else-if type="book thesis" match="any">
+        <text variable="number-of-pages"/>
+        <label plural="never" variable="page" form="short"/>
       </else-if>
-      <else-if type="chapter" match="any">
-        <label plural="never" suffix=" " variable="page" form="short"/>
+      <else-if type="report" match="any">
         <choose>
-          <if match="any" variable="collection-title">
-            <text variable="page" suffix=", "/>
+          <if match="any" variable="container-title">
+            <label plural="never" suffix=". " variable="page" form="short"/>
+            <text variable="number-of-pages"/>
           </if>
           <else>
-            <text variable="page" suffix="."/>
+            <text variable="number-of-pages"/>
+            <label plural="never" variable="page" form="short"/>
           </else>
         </choose>
       </else-if>
-      <else-if type="paper-conference report" match="any">
+      <else>
         <choose>
           <if match="any" variable="container-title">
-            <label plural="never" suffix=" " variable="page" form="short"/>
-            <choose>
-              <if match="any" variable="collection-title">
-                <text variable="page" suffix=", "/>
-              </if>
-              <else>
-                <text variable="page" suffix="."/>
-              </else>
-            </choose>
+            <label plural="never" suffix=". " variable="page" form="short"/>
```

Comment:
for consistency, remove the periods in the suffix here. It's not doing anything (since it's already in the term) and  you don't have it elsewhere.